velox/substrait/tests/VeloxSubstraitRoundTripTest.cpp

```diff
@@ -441,6 +443,21 @@ TEST_F(VeloxSubstraitRoundTripTest, arrayLiteral) {
       "array[], array[array[1,2,3], array[4,5]]");
 }
 
+TEST_F(VeloxSubstraitRoundTripTest, date1) {
+  auto a = makeFlatVector<int32_t>({0, 1});
+  auto b = makeFlatVector<double_t>({0.905791934145, 0.968867771124});
+  // auto resultVec = setVectorFromVariants(
+  //     DATE(), {variant(Date(9020)), variant(Date(8875))}, pool_.get());
+  // auto c = resultVec->asFlatVector<Date>();
+  auto c = makeFlatVector<Date>({Date(8035), Date(8036)});
+
+  auto vectors = makeRowVector({"a", "b", "c"}, {a, b, c});
+  createDuckDbTable({vectors});
+
+  auto plan = PlanBuilder().values({vectors}).planNode();
```

Comment:
Maybe test some queries with DATE literals.

Reply:
Sure. I will add some.

Reply:
I tried something like `.project("c+1")..` or `project("c > 1992-01-01")..`, but seems like these kernels are not supported. 
But I tested the sorting, it did work roundtrip, but I haven't yet checked the intermediate step. 

Do you have anything specific or more suitable case?

Reply:
> I tried something like .project("c+1").. or project("c > 1992-01-01").., but seems like these kernels are not supported.

What happened? You may need to fix the expressions themselves so that they become valid SQL.

Reply:
It said kernels are not supported.

Reply:
Would you share the full error message?

Reply:
```
auto a = makeFlatVector<int32_t>({0, 1});
  auto b = makeFlatVector<double_t>({0.905791934145, 0.968867771124});
  auto c = makeFlatVector<Date>({Date(8036), Date(8035)});

  auto vectors = makeRowVector({"a", "b", "c"}, {a, b, c});
  createDuckDbTable({vectors});

  auto plan = PlanBuilder()
                  .values({vectors})
                  .project({"c + 1"})
                  .orderBy({"c ASC NULLS FIRST"}, false)
                  .planNode();
```

Error

```bash
[ RUN      ] VeloxSubstraitRoundTripTest.date1
E0127 10:08:36.895841 143334 Exceptions.h:68] Line: /home/asus/github/fork/velox/velox/parse/TypeResolver.cpp:111, Function:resolveScalarFunctionType, Expression:  Scalar function signature is not supported: plus(DATE, BIGINT). Supported signatures: (bigint,bigint) -> bigint, (date,interval day to second) -> date, (integer,integer) -> integer, (smallint,smallint) -> smallint, (tinyint,tinyint) -> tinyint, (real,real) -> real, (double,double) -> double, (DECIMAL(a_precision,a_scale),DECIMAL(b_precision,b_scale)) -> DECIMAL(r_precision,r_scale)., Source: USER, ErrorCode: INVALID_ARGUMENT
unknown file: Failure
C++ exception with description "Exception: VeloxUserError
Error Source: USER
Error Code: INVALID_ARGUMENT
Reason: Scalar function signature is not supported: plus(DATE, BIGINT). Supported signatures: (bigint,bigint) -> bigint, (date,interval day to second) -> date, (integer,integer) -> integer, (smallint,smallint) -> smallint, (tinyint,tinyint) -> tinyint, (real,real) -> real, (double,double) -> double, (DECIMAL(a_precision,a_scale),DECIMAL(b_precision,b_scale)) -> DECIMAL(r_precision,r_scale).
Retriable: False
Function: resolveScalarFunctionType
File: /home/asus/github/fork/velox/velox/parse/TypeResolver.cpp
Line: 111
Stack trace:
Stack trace has been disabled.Use --velox_exception_user_stacktrace=true to enable it.
" thrown in the test body.
[  FAILED  ] VeloxSubstraitRoundTripTest.date1 (9 ms)

```

Reply:
@vibhatha Try this

```
  plan = PlanBuilder()
             .values({vectors})
             .filter({"c > DATE '1992-01-01'"})
             .planNode();
  assertPlanConversion(plan, "SELECT * FROM tmp WHERE c > DATE '1992-01-01'");
```